gadgets/src/less_than.rs

```diff
@@ -61,6 +65,7 @@ impl<F: Field, const N_BYTES: usize> LtChip<F, N_BYTES> {
         let lt = meta.advice_column();
         let diff = [(); N_BYTES].map(|_| meta.advice_column());
         let range = pow_of_two(N_BYTES * 8);
+        let u8 = meta.fixed_column();
```

Comment:
It looks like [the first](https://github.com/privacy-scaling-explorations/halo2/blob/fcdd5b910d92ab278d679cf8aa176c885455aa6c/halo2_proofs/src/plonk/circuit.rs#L1891) API returns a `Column<Fixed>` while [the second](https://github.com/privacy-scaling-explorations/halo2/blob/fcdd5b910d92ab278d679cf8aa176c885455aa6c/halo2_proofs/src/plonk/circuit.rs#L1856) returns a`TableColumn`, which is a wrapper around a fixed column.

From the [Halo2 Discord](https://discord.com/channels/995022112811135148/1005560452446683186/1085076854874054806) I read that "halo2 fills in TableColumns with a specified default value". It would be interesting to understand the differences between the two further.

I also saw the range check [being performed as you suggest](https://github.com/icemelon/halo2-examples/blob/master/src/range_check/example2/table.rs), but eventually I opted for this syntax to maintain coherence with [the example](https://github.com/privacy-scaling-explorations/zkevm-circuits/issues/916#issuecomment-1508209602) that @ed255 linked in the issue.